operator/src/main/java/oracle/kubernetes/operator/makeright/MakeRightDomainOperationImpl.java

```diff
@@ -258,7 +260,11 @@ private boolean isForDomain(ClusterResource clusterResource, DomainPresenceInfo
   }
 
   private Step createDomainDownPlan() {
-    return Step.chain(new DownHeadStep(), new DeleteDomainStep(), new UnregisterStep());
+    Step eventStep = null;
+    if (Optional.ofNullable(eventData).map(e -> e.getItem() != DOMAIN_DELETED).orElse(true)) {
+      eventStep = EventHelper.createEventStep(new EventHelper.EventData(DOMAIN_DELETED));
+    }
+    return Step.chain(eventStep, new DeleteDomainStep(), new DownHeadStep(), new UnregisterStep());
```

Comment:
is it intentional to swap the order of DeleteDomainStep and DownHeadStep?

Reply:
Yes. The DownHeadStep unregisters the server status readers. I want to keep it running during the DomainDelete step to update the latest server state in DPI.

Reply:
I updated the step names in  [d0f2eb3](https://github.com/oracle/weblogic-kubernetes-operator/pull/3846/commits/d0f2eb356d1d7b4ad10a5947bf7873bebb0dcbca) to avoid the confusion. Please let me know if this makes sense.